pkg/driver/node/mounter/pod_mounter.go

```diff
@@ -0,0 +1,403 @@
+	// ensure we don't leave `target` mounted if Mountpoint is not started to serve requests for it.
+	unmount := true
+	defer func() {
+		if unmount {
+			if err := pm.unmountTarget(target); err != nil {
+				klog.V(4).ErrorS(err, "Failed to unmount mounted target %s\n", target)
+			} else {
+				klog.V(4).Infof("Target %s unmounted successfully\n", target)
+			}
+		}
+	}()
+
+	// This function can either fail or successfully send mount options to Mountpoint Pod - in which
+	// Mountpoint Pod will get its own fd referencing the same underlying file description.
+	// In both case we need to close the fd in this process.
+	defer pm.closeFUSEDevFD(fuseDeviceFD)
+
+	// Remove old mount error file if exists
+	_ = os.Remove(podMountErrorPath)
+
+	klog.V(4).Infof("Sending mount options to Mountpoint Pod %s on %s", pod.Name, podMountSockPath)
+
+	err = mountoptions.Send(ctx, podMountSockPath, mountoptions.Options{
+		Fd:         fuseDeviceFD,
+		BucketName: bucketName,
+		Args:       args.SortedList(),
+		Env:        env.List(),
+	})
+	if err != nil {
+		klog.V(4).Infof("Failed to send mount option to Mountpoint Pod %s for %s: %s\n", pod.Name, target, err)
+		return fmt.Errorf("Failed to send mount options to Mountpoint Pod %s for %s: %w", pod.Name, target, err)
+	}
+
+	err = pm.waitForMount(ctx, target, pod.Name, podMountErrorPath)
+	if err != nil {
+		return fmt.Errorf("Failed to wait for Mountpoint Pod %s to be ready for %s: %w", pod.Name, target, err)
+	}
+
+	// Mountpoint successfully started, so don't unmount the filesystem
+	unmount = false
+	return nil
+}
+
+// Unmount unmounts the mount point at `target` and cleans all credentials.
+func (pm *PodMounter) Unmount(ctx context.Context, target string, credentialCtx credentialprovider.CleanupContext) error {
+	podID, volumeID := credentialCtx.PodID, credentialCtx.VolumeID
+
+	// TODO: If `target` is a `systemd`-mounted Mountpoint, this would return an error,
+	// but we should still unmount it and clean the credentials.
+	_, podPath, err := pm.waitForMountpointPod(ctx, podID, volumeID)
+	if err != nil {
+		return fmt.Errorf("Failed to wait for Mountpoint Pod for %q: %w", target, err)
+	}
+
+	credentialCtx.WritePath = pm.credentialsDir(podPath)
+
+	// Write `mount.exit` file to indicate Mountpoint Pod to cleanly exit.
+	podMountExitPath := mppod.PathOnHost(podPath, mppod.KnownPathMountExit)
+	_, err = os.OpenFile(podMountExitPath, os.O_RDONLY|os.O_CREATE, credentialprovider.CredentialFilePerm)
+	if err != nil {
+		return fmt.Errorf("Failed to send a exit message to Mountpoint Pod for %q: %w", target, err)
+	}
+
+	err = pm.unmountTarget(target)
+	if err != nil {
+		return fmt.Errorf("Failed to unmount %q: %w", target, err)
+	}
+
+	err = pm.credProvider.Cleanup(credentialCtx)
+	if err != nil {
+		klog.V(4).Infof("Unmount: Failed to clean up credentials for %s: %v", target, err)
+		return fmt.Errorf("Failed to clean up credentials for %q: %w", target, err)
```

Comment:
Same here, why is the format string here different between these log lines?

Reply:
Replied in the other thread